pkg/engine/plan.go

```diff
@@ -242,18 +243,22 @@ func (res *planResult) Close() error {
 }
 
 // printPlan prints the plan's result to the plan's Options.Events stream.
-func printPlan(ctx *Context, result *planResult, dryRun bool) (ResourceChanges, error) {
-	result.Options.Events.preludeEvent(dryRun, result.Ctx.Update.GetTarget().Config)
+func printPlan(ctx *Context, planResult *planResult, dryRun bool) (ResourceChanges, *result.Result) {
+	planResult.Options.Events.preludeEvent(dryRun, planResult.Ctx.Update.GetTarget().Config)
 
 	// Walk the plan's steps and and pretty-print them out.
-	actions := newPlanActions(result.Options)
-	if err := result.Walk(ctx, actions, true); err != nil {
-		return nil, errors.New("an error occurred while advancing the preview")
+	actions := newPlanActions(planResult.Options)
+	if res := planResult.Walk(ctx, actions, true); res != nil {
+		if res.Error() == nil {
+			return nil, res
+		}
```

Comment:
here's an example of how i'm 'checking for bail' @swgillespie .   I would prefer to have a more idiomatic way for this.  Thoughts?